### PR TITLE
Infrastructure to install and run

### PR DESCRIPTION
**Pull request by adamwight**
Still not working, but this gets us to a point where we can run the server.

### CODE REVIEW

File: jade/jade.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    jade [-h | --help]
+    jade server
+
+Available commands:
+    server      Start the web server.
+
+Options:
+    --help      Show this usage message.
+"""
+import docopt_subcommands as dsc
```
Comment (halfak) [inline jade/jade.py:14]: Holy moly

File: jade/trusted_clients.py [hunk 2]
```diff
@@ -29,7 +19,7 @@ def authorize_and_decode(self, request):
             return json2multidict(jwt.decode(
                 encoded_values, self.key_secrets[auth_key],
                 algorithms=[HASH_ALGORITHM]))
-        except JWT_ERRORS as e:
+        except jwt.exceptions.JWTException as e:
```
Comment (halfak) [inline jade/trusted_clients.py:22]: Nice

File: jade/wsgi/server.py [hunk 3]
```diff
@@ -30,15 +31,16 @@ def configure(config):
...
+    state_store = None
 
     bp = routes.configure(
         config, bp, trusted_clients, centralauth, state_store)
     app.register_blueprint(bp)
 
     # Configure swagger-ui routes
     swagger_bp = flask_swaggerui.build_static_blueprint(
-        'ores-swaggerui', __name__,
+        'jade-swaggerui', __name__,
```
Comment (halfak) [inline jade/wsgi/server.py:43]: Nice catch

File: jade/jade.py (hunk 1 above)
Comment (adamwight) [reply jade/jade.py:14]: Yeah... this is not something I'm terribly proud of relying on, unless we're going to make some upstream changes.  But it was expedient, and has the nice property of collapsing subcommand help until it's specifically asked for.

File: jade/applications/wsgi.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+from jade.wsgi import server
+
+from .util import build_config, configure_logging
+
+
+@dsc.command()
+def server_command(args):
+    """
+    usage:
+        {program} server (-h | --help)
```
Comment (adamwight) [inline jade/applications/wsgi.py:11]: @halfak What do you think about this?  I'm torn between consistency across our tools ("applications.wsgi") and having a more human-readable command name.  The user doesn't care that this is an "application" nor that it uses "wsgi", only that it serves REST stuff...

**Pull request merged**